text-input.md

```diff
@@ -0,0 +1,93 @@
+---
+layout: page
+title: Text Input
+sidebar: home_sidebar
+permalink: /text-input/
+---
+
+* TOC Placeholder
+{:toc}
+
+## Introduction
+
+This page describe how to setup basic text input for Flutter applications.
+
+The
+[`Input`](https://docs.flutter.io/flutter/material/Input-class.html)
+widget under
+[Material](https://docs.flutter.io/flutter/material/material-library.html )
+will probably satisfy most text-input use cases. However, `Input` is coupled
+with the Material Design style so there might be cases where you want to use  [`RawInputLine`](https://docs.flutter.io/flutter/widgets/RawInputLine-class.html)
+to achieve desired interfaces outside of the Material Design language.
+
+
+Using the Input Widget
+---------
+
+Assuming that the input is already in focus, the basic data flow for retrieving
+user input will be:
+
+1. User begins to type into the `Input` widget
+2. The `OnChanged` callback is called which passed in the current 'value' of the
+ input
+3. Perform any necessary logic/validation on the current input value
+4. Update the `Input` widget accordingly.
```

Comment:
What does this work here? The user? The widget? What does it mean to "update the input widget accordingly" ?